templates/build_script_python.sh

```diff
@@ -21,77 +21,97 @@
-        echo "$PACKAGE_URL $PACKAGE_NAME"
-        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | GitHub  | Pass |  Both_Install_and_Test_Success"
+fi
+
+if [ -f tox.ini ]; then
+    if !  python3 -m tox -e py39; then
+        echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"   "$WORKDIR/output/test_fails"
+        echo "$PACKAGE_NAME | $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | $SOURCE | Fail | Install_success_but_test_Fails"  
         exit 0
-        fi
+    else
+        echo "------------------$PACKAGE_NAME:install_and_test_both_success-------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"   "$WORKDIR/output/test_success"
+        echo "$PACKAGE_NAME | $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | $SOURCE | Pass | Both_Install_and_Test_Success" 
+        exit 1
+    fi
+elif [ -f noxfile.py ]; then
+    if !  python3 -m nox; then
+        echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"   "$WORKDIR/output/test_fails"
+        echo "$PACKAGE_NAME | $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | $SOURCE | Fail | Install_success_but_test_Fails"  
+        exit 0
+    else
+        echo "------------------$PACKAGE_NAME:install_and_test_both_success-------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"   "$WORKDIR/output/test_success"
+        echo "$PACKAGE_NAME | $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | $SOURCE | Pass | Both_Install_and_Test_Success"   
+        exit 1
+    fi
 else
-        echo "tox.ini not present"
-        if ! pytest ; then
-        echo "------------------$PACKAGE_NAME:Install_success_but_test_fails---------------------"
-        echo "$PACKAGE_URL $PACKAGE_NAME"
-        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | GitHub | Fail |  Install_success_but_test_Fails"
-        exit 2
-        else
-        echo "------------------$PACKAGE_NAME:Install_&_test_both_success-------------------------"
-        echo "$PACKAGE_URL $PACKAGE_NAME"
-        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | GitHub  | Pass |  Both_Install_and_Test_Success"
+    if ! python3 -m pytest; then
+        echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"   "$WORKDIR/output/test_fails"
+        echo "$PACKAGE_NAME | $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | $SOURCE | Fail | Install_success_but_test_Fails"  
         exit 0
-        fi
-
+    else
+        echo "------------------$PACKAGE_NAME:install_and_test_both_success-------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"   "$WORKDIR/output/test_success"
+        echo "$PACKAGE_NAME | $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | $SOURCE | Pass | Both_Install_and_Test_Success"   
+        exit 1
```

Comment:
Exit status are incorrect it should be as follow:
If package installation fails: exit 1
if package test fails: exist 2
If both package installation and package test success: exit 0